zips/zip-0234.md

```diff
@@ -56,83 +56,55 @@ network's economic model, potentially impacting its security and stability.
-disbursed after they are deposited back into the ZSF.
+This new ZSF-based issuance scheme solves these problems by ensuring a more
+consistent and predictable rate of coin issuance, whilst preserving the core
+aspects of Zcash's issuance policy. At the same time, it introduces the
+first mechanism for payout from the Sustainability Fund (ZSF), essentially
+distributing ZSF deposits and unclaimed transaction fees across all future block
+subsidies.
 
 # Requirements
 
 Smoothing the issuance curve is possible using an exponential decay formula
 that satisfies the following requirements:
 
-## Issuance Requirements
-
 1. The issuance can be summarised into a reasonably simple explanation
 2. Block subsidies approximate a continuous function
 3. If there are funds in the ZSF, then the block subsidy must be non-zero,
    preventing any final “unmined” zatoshis
 4. For any 4 year period, all paid out block subsidies are approximately equal
    to half of the ZSF at the beginning of that 4 year period, if there are no
    deposits into the ZSF during those 4 years
+5. Decrease the short-term impact of the deployment of this ZIP on block reward
+   recipients, and minimise the potential reputation risk to Zcash of changing
+   the block reward amount.
 
-TODO daira: add a requirement that makes the initial total issuance match the previous total issuance  
+TODO daira: add a requirement that makes the initial total issuance match the previous total issuance
 
 # Specification
 
-## Goals
-
-We want to decrease the short-term impact of the deployment of this ZIP on
-block reward recipients, and minimise the potential reputational risk to Zcash
-of changing the block reward amount.
+## Definitions
 
-## Constants
+`BLOCK_SUBSIDY_FRACTION` = 4126 / 10,000,000,000 or `0.0000004126`
 
-Define constants:
+`DEPLOYMENT_BLOCK_HEIGHT` = 2726400
 
-“`BLOCK_SUBSIDY_FRACTION`” = 4126 / 10,000,000,000 or `0.0000004126`
+`ZSF_BALANCE` = `MAX_MONEY - ChainValue`, where `ChainValue` includes all ZEC available on the network at a given block height, across all value pools
 
-"`DEPLOYMENT_BLOCK_HEIGHT`" = 2726400
+`ZsfBalanceAfter(block_height)` = `ZSF_BALANCE` as calculated above the specified block height
 
 ## Issuance Calculation
 
 At the `DEPLOYMENT_BLOCK_HEIGHT`, nodes should switch from the current issuance
 calculation, to the following:
 
-Given the block height `h` define a function **BlockSubsidy(h)**, such that:
-
-**BlockSubsidy(h)** = Block subsidy for a given `h`, that satisfies above
-requirements.
-
-Using an exponential decay function for **BlockSubsidy** satisfies requirements
-**R1** and **R2** above:
-
-`BlockSubsidy(h) = BLOCK_SUBSIDY_FRACTION * ZsfBalanceAfter(h - 1)`
-
-Finally, to satisfy **R3** above we always round up to the next zatoshi.
-
-`BlockSubsidy(h) = ceiling(BLOCK_SUBSIDY_FRACTION * ZsfBalanceAfter(h - 1))`
+`BlockSubsidy(block_height) = ceiling(BLOCK_SUBSIDY_FRACTION * ZsfBalanceAfter(block_height - 1))`
```

Comment:
This change as it's currently specified would require Zebra to defer validating a block's coinbase transaction value balances and funding stream outputs until a block at the previous height has been validated.

Using a fraction of the `ZsfBalanceAfter(block_height - C)` instead would help Zebra to continue validating many coinbase transactions in parallel, and would reduce implementation effort as it won't require that that logic be moved to the state service's write block task.

```suggestion
`BlockSubsidy(block_height) = ceiling(BLOCK_SUBSIDY_FRACTION * ZsfBalanceAfter(block_height - 1000))`
```

Reply:
This is a really helpful insight; thanks. The idea to introduce delay to the issuance curve makes a lot of sense to me.